src/lib/scheduler/operator_task.cpp

```diff
@@ -30,20 +30,24 @@ std::vector<std::shared_ptr<AbstractTask>> OperatorTask::make_tasks_from_operato
 std::shared_ptr<AbstractTask> OperatorTask::_add_tasks_from_operator(
     const std::shared_ptr<AbstractOperator>& op, std::vector<std::shared_ptr<AbstractTask>>& tasks,
     std::unordered_map<std::shared_ptr<AbstractOperator>, std::shared_ptr<AbstractTask>>& task_by_op) {
+  // Early out: We do not want to re-run operators if they have already been executed by previous tasks.
+  if (op->executed()) return nullptr;
```

Comment:
Changed to 

```cpp
  // Early out: We do not want to schedule operators for execution that have already been executed by previous tasks
  //            because the results should already be accessible.
  if (op->executed()) {
    Assert(op->get_output(), "Expected executed operator to return an output.");
    return nullptr;
  }
```

Reply:
> Won't this be illegal with the recent changes? Turn it into an assertion?

No. For example.

1. Operator A gets scheduled for execution.
2. Operator A gets executed.
3. An uncorrelated subquery is executed during runtime by the ExpressionEvaluator:
```cpp
  const auto tasks = OperatorTask::make_tasks_from_operator(row_pqp);
  Hyrise::get().scheduler()->schedule_and_wait_for_tasks(tasks);
```
`row_pqp` contains a reference to Operator A.

Nothing would go wrong if we scheduled Operator A another time because there is an early out in `execute()` anyway. But, why should we schedule tasks from which we know they are pointless?.`

Reply:
(We decided to remove this shortcut to avoid further confusion)